azurerm/resource_arm_hdinsight.go

```diff
@@ -0,0 +1,888 @@
+				//TODO:ValidateFunc: validateHDInsightKind,
+			},
+			"gateway": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+			},
+			"cluster_identity": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+			},
+			"core_site": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+			},
+			"security": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"domain": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"organizational_unit_dn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ldaps_urls": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ldaps_url": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+							Set: resourceArmHDInsightldapsHash,
+						},
+						"domain_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"domain_userpassword": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"cluster_users_group_dns": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+
+									"users_group_dn": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+							Set: resourceArmHDInsightUserGroupDnHash,
+						},
+					},
+				},
+				Set: resourceArmHDInsightSecurityHash,
+			},
+			"node_role": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"count": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      1,
+							ValidateFunc: validateArmContainerServiceAgentPoolProfileCount,
+						},
+						"size": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"admin_password": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key_data": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"vnet_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"subnet_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
```

Comment:
can we split these two out into their own `network_profile` block to match the Azure API? e.g.

```
"network_profile": {
  Type:     schema.TypeList,
  Optional: true,
  MaxItems: 1,
  Elem: &schema.Resource{
    Schema: map[string]*schema.Schema{
      "virtual_network_id": {
        Type:     schema.TypeString,
        Required: true,
      },
      "subnet_name": {
        Type:     schema.TypeString,
        Required: true,
      },
    },
  },
},
```